executor/runtime/docker/docker_isolate.go

```diff
@@ -35,6 +35,15 @@ func waitForTitusIsolateWithHost(parentCtx context.Context, taskID, host string,
 			return false
 
 		case <-ticker.C:
+			//give priority to a possible concurrent Done() event non-blocking way
+			// This is the only way to provide "priority" to our done messages
+			// otherwise we cannot control if the outer select picks them over the ticker
+			select {
```

Comment:
You should be able to just check ctx.Err() here instead of doing a full select.

Reply:
Ah, that is slightly cleaner